client/src/app/site/agenda/components/list-of-speakers/list-of-speakers.component.ts

```diff
@@ -297,14 +312,37 @@ export class ListOfSpeakersComponent extends BaseViewComponent implements OnInit
     }
 
     /**
-     * React to manual in the sorting order.
-     * Informs the repo about changes in the order
-     * @param listInNewOrder Contains the newly ordered list of ViewSpeakers
+     * When entering sort mode save the old speaker order to restore on cancel
+     * save in the form of (id => position)
      */
-    public onSortingChange(listInNewOrder: ViewSpeaker[]): void {
-        // extract the ids from the ViewSpeaker array
-        const userIds = listInNewOrder.map(speaker => speaker.id);
-        this.listOfSpeakersRepo.sortSpeakers(this.viewListOfSpeakers, userIds).then(null, this.raiseError);
+    public onEnterSorting(): void {
+        this.oldSpeakerOrder = [];
+        this.speakers.forEach((speaker, idx) => {
```

Comment:
Please put the store/restore capability directly into the sorting list component, e.g. a `store()` method, storing the current `this.array` values (in the sorting list component) into an object. on `restore()` the sorted order is redone.